### DISCUSSION TITLE
Sub-path routing with country and language

### DISCUSSION THREAD

**Discussion by noidar** [vercel/next.js#36397] [Help] [answered]
I need to implement a translation soluition for a Nextjs project that should handle http://localhost:3000/de/ch/slugname
language:de country:ch
instead I am receiving { country:'slugname ', language:'ch'}
Here is an example of what I tried so far: https://github.dev/noidar/next-i18next-sub-path-routing-test

Is there any solution? Thanks.

**Comment by vadymshymko** [comment, marked as answer]
Hi. I don't think you need the language folder in pages. It can / should be removed, and to get the language inside the components, use the following code:
```
import {useRouter} from 'next/router'

const router = useRouter();
const language = router.locale;
```

**Reply by vadymshymko** [reply]
And you can read more about nextjs i18n routing [here](https://nextjs.org/docs/advanced-features/i18n-routing).

**Reply by noidar** [reply]
> Hi. I don't think you need the language folder in pages. It can / should be removed, and to get the language inside the components, use the following code:
> 
> ```
> import {useRouter} from 'next/router'
> 
> const router = useRouter();
> const language = router.locale;
> ```

Hi Vadim, thank you for your answer, I will try that!
